sphinx/util/__init__.py

```diff
@@ -245,18 +245,17 @@ def get_module_source(modname: str) -> Tuple[str, str]:
         raise PycodeError('error importing %r' % modname, err)
     filename = getattr(mod, '__file__', None)
     loader = getattr(mod, '__loader__', None)
+    try:  # prefer Native loader, as it respects #coding directive
+        filename = loader.get_source(modname)
+        if filename:
+            return 'string', filename
+    except Exception as err:
```

Comment:
document says `loader.get_source()` raises ImportError only. So this should handle it.
https://docs.python.org/3/library/importlib.html#importlib.abc.InspectLoader.get_source

Reply:
Ok. I`ll update it